scc-big-data.ipynb

```diff
@@ -0,0 +1,1883 @@
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "c9ba4221-076c-4c6a-bfef-96c2901e9e5b",
+   "metadata": {
+    "nbgrader": {
+     "cell_type": "code",
+     "checksum": "9505e2598968c0493f6dff974fa25204",
+     "grade": false,
+     "grade_id": "census-ans",
+     "locked": false,
+     "schema_version": 3,
+     "solution": true,
+     "task": false
+    },
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "#  Download the Census data for the state of Illinois for the year 2021\n",
+    "\n",
+    "census_data_path = os.path.join(proj_dir, 'census_data', 'il_census.csv')\n",
+    "if not os.path.exists(census_data_path):\n",
+    "    api_key = getpass.getpass('U.S. Census API Key')\n",
+    "    c = Census(api_key)\n",
+    "    il_census = c.acs5.state_county_tract(fields = ('NAME', 'B06011_001E', 'B19301_001E', 'B01003_001E', 'B25035_001E'),\n",
+    "                                      state_fips = states.IL.fips,\n",
+    "                                      county_fips = \"*\",\n",
+    "                                      tract = \"*\",\n",
+    "                                      year = 2021) \n",
+    "    il_df = pd.DataFrame(il_census)\n",
+    "    print(\"Column data types for il_df data:\\n{}\".format(il_df.dtypes))\n",
+    "    il_df.to_csv(census_data_path, index=False, header=False)\n",
+    "    \n",
+    "# Explicitly set data types to object to preserve leading zeros\n",
+    "column_names = (['NAME', 'median_income', 'per_capita_income',\n",
+    "                     'population', 'median_year_built','state', 'county', 'tract', 'GEOID'])\n",
+    "column_dtypes = ({'NAME': object, 'median_income': float, 'per_capita_income': float,\n",
+    "                      'population': float, 'median_year_built': int, 'state': object,\n",
+    "                      'county': object, 'tract': object, 'GEOID': object})\n",
+    "    \n",
+    "# Read the CSV file into a DataFrame\n",
+    "il_census_csv = pd.read_csv(census_data_path, names=column_names, dtype=column_dtypes)\n",
+    "  \n",
+    "# Show the dataframe\n",
+    "print(il_census_csv.tail(10))\n",
+    "print('Shape: ', il_census_csv.shape)\n",
+    "#  Confirm that the data is in the same format after changing it to .csv file\n",
+    "print(\"Column data types for il_census_csv data:\\n{}\".format(il_census_csv.dtypes))\n",
+    "#print(\"Column data types for il_df data:\\n{}\".format(il_df.dtypes))\n",
+    "\n",
+    "\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "fb106d60-4b9f-418e-8d96-4a6ecfeb03e1",
+   "metadata": {
+    "deletable": false,
+    "nbgrader": {
+     "cell_type": "code",
+     "checksum": "9505e2598968c0493f6dff974fa25204",
+     "grade": false,
+     "grade_id": "census-ans",
+     "locked": false,
+     "schema_version": 3,
+     "solution": true,
+     "task": false
+    },
+    "tags": []
+   },
+   "source": [
+    "#### For what ever reason, I can't get this cell to delete"
```

Comment:
I had this issue before and I had to overwrite the cell's metadata like:
 "metadata": {"deletable": true} from the original false setting.